src/ansible/provisioning.rs

```diff
@@ -923,8 +1006,40 @@ impl AnsibleProvisioner {
     ) -> Result<()> {
         let start = Instant::now();
 
-        let sk_map = self
-            .deposit_funds_to_uploaders(&FundingOptions {
+        let sk_map = if let Some(wallet_keys) = &options.wallet_secret_keys {
```

Comment:
Could we move this logic into `funding.rs` maybe? Just to keep everything related to funding at a single place?